langserver/server.go

```diff
@@ -164,6 +169,29 @@ func (s *server) connectPrometheus(url string) error {
 			return err
 		}
 
+		bodyBytes, _ := ioutil.ReadAll(resp.Body)
+		jsonResponse := map[string]map[string]string{}
```

Comment:
You could unmarshal into [this type](https://github.com/prometheus/prometheus/blob/bc703b64568ebfaecf27b9b70be737ad318e217a/web/api/v1/api.go#L119) and store the result as a member of the langserver struct.

In this case it's better to not import the package but copy over the struct def (with a notice where it's from) since we would drag in to many dependencies otherwise.

Reply:
You also need to wrap it into [this](https://github.com/prometheus/prometheus/blob/bc703b64568ebfaecf27b9b70be737ad318e217a/web/api/v1/api.go#L144) type.

Reply:
Sorry for the inconvenience. I resolved the error with type insertion

Reply:
That should work, too.